upup/pkg/fi/cloudup/dns.go

```diff
@@ -216,3 +196,36 @@ func precreateDNS(cluster *api.Cluster, cloud fi.Cloud) error {
 
 	return nil
 }
+
+// buildPrecreateDNSHostnames returns the hostnames we should precreate
+func buildPrecreateDNSHostnames(cluster *api.Cluster) []string {
+	dnsInternalSuffix := ".internal." + cluster.ObjectMeta.Name
+
+	var dnsHostnames []string
+
+	if cluster.Spec.MasterPublicName != "" {
+		dnsHostnames = append(dnsHostnames, cluster.Spec.MasterPublicName)
+	} else {
+		glog.Warningf("cannot pre-create MasterPublicName - not set")
+	}
+
+	if cluster.Spec.MasterInternalName != "" {
+		dnsHostnames = append(dnsHostnames, cluster.Spec.MasterInternalName)
+	} else {
+		glog.Warningf("cannot pre-create MasterInternalName - not set")
```

Comment:
We log it as a warning.  We certainly shouldn't fail everything ... we might well not have permissions on this IAM account or something.  Warning seems appropriate given we aren't stopping.